test/conformance/blue_green_test.go

```diff
@@ -107,9 +107,22 @@ func TestBlueGreenRoute(t *testing.T) {
 		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
 	}
 
-	// TODO(tcnghia): Add a RouteStatus method to create this string.
-	blueDomain := fmt.Sprintf("%s.%s", blue.TrafficTarget, route.Status.Domain)
-	greenDomain := fmt.Sprintf("%s.%s", green.TrafficTarget, route.Status.Domain)
+	var blueDomain, greenDomain string
+	for _, tt := range route.Status.Traffic {
+		if tt.Name == blue.TrafficTarget {
+			// Strip prefix as WaitForEndPointState expects a domain
+			// without scheme.
+			blueDomain = strings.TrimPrefix(tt.URL, "http://")
+		}
+		if tt.Name == green.TrafficTarget {
+			// Strip prefix as WaitForEndPointState expects a domain
+			// without scheme.
+			greenDomain = strings.TrimPrefix(tt.URL, "http://")
+		}
+	}
+	if blueDomain == "" || greenDomain == "" {
```

Comment:
Do we want to validate the actual values?

Reply:
For conformance I don't want to validate how the sub-route domains get formed (i.e. "green.ksvc.domain.com" or "green-ksvc.domain.com").

I could use url.Parse() and make sure a domain comes out, but I am not in love with the validation, and we do not do anything similar for `route.Status.Domain` in any of our e2e tests.

If the url is not empty and the domain is invalid the problem should be fairly easy to spot with existing output. It should look something like:

`util.go:210: For domain green.blue-green-route-nfluhlbe.serving-tests.example.com: wanted at least 50, got 0 requests.`